vega_sim/scenario/hedged_market_maker/scenario.py

```diff
@@ -0,0 +1,333 @@
+import numpy as np
+from typing import Any, Callable, List, Optional
+from vega_sim.environment.agent import Agent
+
+from vega_sim.api.market import MarketConfig
+
+from vega_sim.scenario.scenario import Scenario
+from vega_sim.scenario.common.utils.price_process import random_walk
+from vega_sim.environment.environment import MarketEnvironmentWithState
+from vega_sim.null_service import VegaServiceNull
+from vega_sim.scenario.hedged_market_maker.agents import (
+    MARKET_CREATOR,
+    MARKET_SETTLER,
+    INT_MARKET_MAKER_KEY_A,
+    INT_MARKET_MAKER_KEY_B,
+    EXT_MARKET_MAKER,
+    AUCTION_PASS_BID,
+    AUCTION_PASS_ASK,
+    INT_RANDOM_TRADER,
+    EXT_RANDOM_TRADER,
+    INT_INFORMED_TRADER,
+)
+from vega_sim.scenario.common.agents import (
+    MarketManager,
+    OpenAuctionPass,
+    HedgedMarketMaker,
+    ExponentialShapedMarketMaker,
+    InformedTrader,
+    MarketOrderTrader,
+)
+
+
+class HedgedMarket(Scenario):
+    def __init__(
+        self,
+        num_steps: int = 290,
+        step_length_seconds: int = 60,
+        block_length_seconds: int = 1,
+        price_process_fn: Optional[Callable[[None], List[float]]] = None,
+        initial_price: Optional[float] = None,
+        asset_name: str = "ETH",
+        asset_adp: int = 5,
+        int_name: str = "ETHUSD (Internal Market)",
+        int_mdp: int = 3,
+        int_pdp: int = 3,
+        int_lock: float = 24 * 60 * 60,
+        ext_name: str = "ETHUSD (External Market)",
+        ext_mdp: int = 3,
+        ext_pdp: int = 3,
+        ext_lock: float = 5 * 60,
+        int_mm_initial_asset_mint: float = 3e5,
+        int_mm_commitment_amount: float = 1e5,
+        int_mm_profit_margin: float = 0.0000,
+        int_mm_max_leverage: float = 10.0,
+        ext_mm_num_levels: int = 25,
+        ext_mm_tick_spacing: float = 0.1,
+        ext_mm_kappa: float = 1,
+        ext_mm_market_kappa: float = 10,
+        int_it_proportion: float = 0.1,
+        int_it_accuracy: float = 1.0,
+        int_it_lookahead: int = 30,
+        random_agent_ordering: bool = False,
+        transactions_per_block: int = 1000,
+        state_extraction_fn: Optional[
+            Callable[[VegaServiceNull, List[Agent]], Any]
+        ] = None,
+        state_extraction_freq: int = 1,
+        pause_every_n_steps: Optional[int] = None,
```

Comment:
If you're feeling brave, we should probably start trying to be better at actually documenting what the million parameters passed into these scenarios actually do 😄

Reply:
Sounds good, do you think it is worth starting to flesh out each scenario with a README? Detailed enough docstrings could start to get very long for some of these scenarios I think.

Reply:
Possibly a good combination would be a module level docstring giving a broad overview of what the scenario runs and does. That way the docstring for the class itself can be a little more terse in how it refers to what args do.